mllib/src/test/scala/org/apache/spark/ml/source/libsvm/LibSVMRelationSuite.scala

```diff
@@ -79,4 +79,23 @@ class LibSVMRelationSuite extends SparkFunSuite with MLlibTestSparkContext {
     val v = row1.getAs[SparseVector](1)
     assert(v == Vectors.sparse(100, Seq((0, 1.0), (2, 2.0), (4, 3.0))))
   }
+
+  test("write libsvm data and read it again") {
+    val df = sqlContext.read.format("libsvm").load(path)
+    val writepath = path + "_2"
+    df.write.save(writepath)
```

Comment:
Could you help check why it passed tests? It could be a Spark SQL feature/bug, e.g., it forces using Parquet format when there are Parquet files on the input path.